x/evm/keeper/gas.go

```diff
@@ -30,21 +31,24 @@ func (k *Keeper) GetEthIntrinsicGas(ctx sdk.Context, msg core.Message, cfg *para
 // consumed in the transaction. Additionally, the function sets the total gas consumed to the value
 // returned by the EVM execution, thus ignoring the previous intrinsic gas consumed during in the
 // AnteHandler.
-func (k *Keeper) RefundGas(ctx sdk.Context, msg core.Message, leftoverGas uint64, denom string) error {
+func (k *Keeper) RefundGas(ctx sdk.Context, msg core.Message, leftoverGas uint64, denom string, denomDec uint32) error {
 	// Return EVM tokens for remaining gas, exchanged at the original rate.
 	remaining := new(big.Int).Mul(new(big.Int).SetUint64(leftoverGas), msg.GasPrice())
 
+	// set the corresponding decimals for the EVM keeper bank module wrapper
+	if err := k.bankWrapper.WithDecimals(denomDec); err != nil {
+		return err
+	}
```

Comment:
Shouldn't be a static value the decimals used in the wrapper? Why are we calling again WithDecimals here?

Reply:
the decimals may change based on the context of a query or tx. For example, a historical query before the upgrade that includes this changes, will not have the decimals set, so it will default to 0